src/masoniteorm/scopes/UUIDPrimaryKeyScope.py

```diff
@@ -23,7 +23,9 @@ def generate_uuid(self, builder, uuid_version):
         return str(uuid_func(*args))
 
     def set_uuid_create(self, builder):
-        uuid_version = getattr(builder._model, "__uuid_version__", 4)
-        builder._creates.update(
-            {builder._model.__primary_key__: self.generate_uuid(builder, uuid_version)}
-        )
+        # if there is already a primary key, no need to set a new one
+        if not builder._model.__primary_key__:
```

Comment:
This conditional is checking the model has a primary key. That is wrong. This should be replaced with:

```suggestion
        if builder._model.__primary_key__ not in builder._creates:
```

Reply:
Thats why the tests are failing